### PR TITLE
Add stepping granularities to the step requests.

### PR DESCRIPTION
**Pull request by auott**
This allows step requests (Next, StepInto, StepOut, StepBack) to take
an optional 'steppingGranularity', which provides a hint for how far
a debug adapter should let the step go.

This provides support for proposal https://github.com/microsoft/debug-adapter-protocol/issues/110

### CODE REVIEW

**Comment by auott** [comment]
@weinand - PR for stepping granularities proposal.

File: debugAdapterProtocol.json [hunk 1]
```diff
@@ -3397,6 +3417,12 @@
 			"required": [ "verified" ]
 		},
 
+		"SteppingGranularity": {
+			"type": "string",
+			"description": "The granularity of the requested step.",
+			"enum": [ "statement", "line", "instruction" ]
```
Comment (weinand) [inline debugAdapterProtocol.json:3423]: Please add descriptions for the individual values (by means of an "enumDescriptions" array).

Comment (auott) [reply debugAdapterProtocol.json]: Added. 

I noticed that the comments weren't getting added to specification.md so I also went ahead and updated the generator to add comments for standalone enum definitions. It added comments on a couple of other places as well. I can also revert those changes if you'd prefer to leave it as is for now (or to handle it separately).

File: specification.md [hunk 2]
```diff
@@ -3421,6 +3446,7 @@ interface FunctionBreakpoint {
 This enumeration defines all possible access types for data breakpoints.
 
 ```typescript
+This enumeration defines all possible access types for data breakpoints.
```
Comment (puremourning) [inline specification.md:3449]: this doesn't seem like valid typescript. Shouldn't this be a comment?

It seems to duplicate the line above...

Comment (auott) [reply specification.md:3449]: Ah, thanks. Generator doesn't behave the way I was expecting.
I'll fix this up.

File: debugAdapterProtocol.json (hunk 1 above)
Comment (auott) [reply debugAdapterProtocol.json]: Reverted for now - was not quite as simple as I initially thought.

File: specification.md (hunk 2 above)
Comment (auott) [reply specification.md]: Reverted those...though new version of generator isn't picking up the enum values so still investigating that.

File: specification.md [hunk 3]
```diff
@@ -3503,6 +3528,15 @@ interface Breakpoint {
 }
 ```
 
+### <a name="Types_SteppingGranularity" class="anchor"></a>SteppingGranularity
+
+The granularity of the requested step.
+
+```typescript
+interface SteppingGranularity {
```
Comment (puremourning) [inline specification.md:3536]: this looks wrong. it's an empty interface ?

Where are the values defined ?

Comment (auott) [reply specification.md:3536]: Per other comment - digging in to this. When I re-ran the generator to remove the comments it no longer has the values.

**Comment by weinand** [comment]
@auott thanks a lot for the PR.
I did minor edits on some comments and added a section to the changelog.
The "enumDescriptions" do not yet show up in the generated markdown (but they do show up in the generated TypeScript).
I will take care of updating the markdown generator...

**Pull request merged**